docs/pages/2020_Intro_Week/sections/workflows_basic.rst

```diff
@@ -1,6 +1,637 @@
+
+Clearly, when writing workflows that involve the use of an *ab initio* code like Quantum ESPRESSO, it is better to use a work chain.
+Below you can find the basic rules that allow you to convert your workfunction-based script to a workchain-based one and a snippet example focusing on the code used to perform the calculation of an equation of state.
+
+.. include:: include/snippets/workflows_eos_workchain.py
+    :code: python
+
+.. warning::
+
+    WorkChains need to be defined in a **separate file** from the script used to run them.
+    E.g. save your WorkChain in ``workchains.py`` and use ``from workchains import MyWorkChain`` to import it in your script.
+
+-   Instead of using decorated functions you need to define a class, inheriting from a prototype class called ``WorkChain`` that is provided by AiiDA in the ``aiida.engine`` module.
+
+-   Within your class you need to implement a ``define`` classmethod that always takes ``cls`` and ``spec`` as inputs.
+    Here you specify the main information on the workchain, in particular:
+
+    -   The *inputs* that the workchain expects.
+        This is obtained by means of the ``spec.input()`` method, which provides as the key feature the automatic validation of the input types via the ``valid_type`` argument.
+        The same holds true for outputs, as you can use the ``spec.output()`` method to state what output types are expected to be returned by the workchain.
+
+    -   The ``outline`` consisting in a list of 'steps' that you want to run, put in the right sequence.
+        This is obtained by means of the method ``spec.outline()`` which takes as input the steps.
+        *Note*: in this example we just split the main execution in two sequential steps, that is, first ``run_eos`` then ``results``.
+        However, more complex logic is allowed, as will be explained in :ref:`another section<2019_sintef_convpressure>`.
+
+-   You need to split your main code into methods, with the names you specified before into the outline (``run_eos`` and ``results`` in this example).
+    Where exactly should you split the code?
+    Well, the splitting points should be put where you would normally block the execution of the script for collecting results in a standard work function, namely whenever you call the method ``.result()``.
+    Each method should accept only one parameter, ``self``, e.g. ``def step_name(self)``.
+
+-   You will notice that the methods reference the attribute ``ctx`` through ``self.ctx``, which is called the *context* and is inherited from the base class ``WorkChain``.
+    A python function or process function normally just stores variables in the local scope of the function.
+    For instance, in the example of :ref:`this subsection<2019_sintef_sync>`, you stored the completed calculations in the ``calculations`` dictionary, that was a local variable.
```

Comment:
as above, reference to another tutorial

Reply:
removed the link since not sure if we will have this advanced section